File: SchoolManagementApi/DTOs/Enrollments/StudentAcademicRecordDto.cs
namespace SchoolManagementApi.DTOs.Enrollments;

public class StudentAcademicRecordDto
{
    public int StudentId { get; set; }
    public string StudentName { get; set; } = string.Empty;
    public List<EnrolledSubjectForRecordDto> Subjects { get; set; } = new();
    public int TotalCredits { get; set; }
}

public class EnrolledSubjectForRecordDto
{
    public int SubjectId { get; set; }
    public string SubjectName { get; set; } = string.Empty;
    public string TeacherName { get; set; } = string.Empty;
    public int Credits { get; set; }
}

File: SchoolManagementApiNode/src/enrollments/dto/student-academic-record.dto.ts
import { ApiProperty } from '@nestjs/swagger';

class EnrolledSubjectForRecordDto {
  @ApiProperty()
  subjectId: number;

  @ApiProperty()
  subjectName: string;

  @ApiProperty()
  teacherName: string;

  @ApiProperty()
  credits: number;
}

export class StudentAcademicRecordDto {
  @ApiProperty()
  studentId: number;

  @ApiProperty()
  studentName: string;

  @ApiProperty({ type: [EnrolledSubjectForRecordDto] })
  subjects: EnrolledSubjectForRecordDto[];

  @ApiProperty()
  totalCredits: number;
}
